projects/dendrites/permutedMNIST/experiments/batch_pmnist.py

```diff
@@ -0,0 +1,147 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+"""
+These are baseline experiments testing how well the sparse network without any dendrites
+performs on a batch version of permuted MNIST. The goal is to see if the base
+feed-forward network has sufficient capacity to learn all num_tasks*60000 training
+examples simultaneously.
+
+This accuracy could be a reasonable upper bound for the continual learning version,
+which has to have learned all the same examples by the end of the experiment.
+"""
+
+import os
+from copy import deepcopy
+
+import numpy as np
+import ray.tune as tune
+import torch
+import torch.nn.functional as F
+
+from nupic.research.frameworks.dendrites import DendriticMLP
+from nupic.research.frameworks.dendrites.dendrite_cl_experiment import (
+    DendriteContinualLearningExperiment,
+)
+from nupic.research.frameworks.dendrites.modules.dendritic_layers import (
+    ZeroSegmentDendriticLayer,
+)
+from nupic.research.frameworks.pytorch.datasets import PermutedMNIST
+from nupic.research.frameworks.vernon import mixins
+
+
+class NoDendriteBatchExperiment(mixins.RezeroWeights,
+                                mixins.PermutedMNISTTaskIndices,
+                                DendriteContinualLearningExperiment):
+
+    def should_stop(self):
+        return self.current_epoch >= self.epochs
+
+    def run_task(self):
+        self.train_loader.sampler.set_active_tasks(range(self.num_tasks))
+        self.val_loader.sampler.set_active_tasks(range(self.num_tasks))
+        self.logger.info("Training task %d, epoch %d...",
+                         self.current_task, self.current_epoch)
+
+        # Just run one epoch
+        ret = self.run_epoch()
```

Comment:
For flexibility, `run_epoch` should be called multiple times, since I find it hard to imagine a single pass through num_tasks * 60k images would be sufficient.

Reply:
Also, the config specifies `epochs=3` below, so that's not being used.

Reply:
I believe it does run it through `epochs` times. `run_epoch()` increments `self.current_epoch`. `run_task()` is just a pass through here to run one epoch and is called repeatedly until `should_stop()`, redefined on line 54, is `True`. So essentially it runs through epochs the same way a supervised experiment does, but retains the rest of the code path of `DendriteContinualLearningExperiment`.

Reply:
If `run_epoch` was called multiple times when you ran this, I'll  believe that. Then, the following line can be placed in `setup_experiment`, and doesn't need to be called repeatedly, right?

```
self.train_loader.sampler.set_active_tasks(range(self.num_tasks))
```